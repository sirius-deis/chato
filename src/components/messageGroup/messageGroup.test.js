import { render, screen } from "@testing-library/react";
import MessageGroup from "./messageGroup";

const messages = [
  {
    id: 1,
    senderId: "1",
    message: "Message 1",
    createdAt: new Date("2024-07-17T21:08:30").toISOString(),
  },
  {
    id: 2,
    senderId: "1",
    message: "Message 2",
    createdAt: new Date("2024-07-17T21:09:50").toISOString(),
  },
];

describe("MessageGroup component", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <MessageGroup messages={messages} userId={1} />
    );
    expect(container).toMatchSnapshot();
  });
  it("should render messages", () => {
    render(<MessageGroup messages={messages} />);
    messages.forEach(({ message }) =>
      expect(screen.getByText(message)).toBeInTheDocument()
    );
  });
});
