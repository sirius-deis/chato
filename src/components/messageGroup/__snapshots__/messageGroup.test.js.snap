// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageGroup component should match snapshot 1`] = `
<div>
  <div
    class="sc-eqUAAy hkHHhd"
    data-testid="message_group"
  >
    <div
      class="sc-gEvEer eJajcW  "
    >
      Message 1
    </div>
    <div
      class="sc-gEvEer eJajcW last "
    >
      Message 2
    </div>
  </div>
</div>
`;
